source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,318 @@
+/* StartConditionsPanel.cpp
+Copyright (c) 2020-2021 by FranchuFranchu <fff999abc999@gmail.com>
+Copyright (c) 2021 by Benjamin Hauch
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "StartConditionsPanel.h"
+
+#include "text/alignment.hpp"
+#include "Command.h"
+#include "ConversationPanel.h"
+#include "text/DisplayText.h"
+#include "FillShader.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "text/Format.h"
+#include "GameData.h"
+#include "Information.h"
+#include "Interface.h"
+#include "text/layout.hpp"
+#include "MainPanel.h"
+#include "PlayerInfo.h"
+#include "Planet.h"
+#include "Preferences.h"
+#include "Rectangle.h"
+#include "ShipyardPanel.h"
+#include "StarField.h"
+#include "StartConditions.h"
+#include "System.h"
+#include "text/truncate.hpp"
+#include "UI.h"
+
+#include <algorithm>
+
+using namespace std;
+
+
+
+StartConditionsPanel::StartConditionsPanel(PlayerInfo &player, UI &gamePanels, const StartConditionsList &scenarios, const Panel *parent)
+	: player(player), gamePanels(gamePanels), parent(parent), scenarios(scenarios), startIt(scenarios.end()),
+	bright(*GameData::Colors().Get("bright")), medium(*GameData::Colors().Get("medium")),
+	selectedBackground(*GameData::Colors().Get("faint")),
+	description(FontSet::Get(14))
+{
+	const Interface *startConditionsMenu = GameData::Interfaces().Find("start conditions menu");
+	if(startConditionsMenu)
+	{
+		// Ideally, we want the content of the boxes to be drawn in Interface.cpp.
+		// However, we'd need a way to specify arbitrarily extensible lists there.
+		// We also would need to a way to specify truncation for such a list.
+		// Such a list would also have to be scrollable.
+		descriptionBox = startConditionsMenu->GetBox("start description");
+		entriesContainer = startConditionsMenu->GetBox("start entry list");
+		entryBox = startConditionsMenu->GetBox("start entry item bounds");
+		entryTextPadding = startConditionsMenu->GetBox("start entry text padding").Dimensions();
+	}
+	
+	// Default the selection to the most recently loaded scenario, if possible.
+	const auto startCount = scenarios.size();
+	if(startCount >= 1)
+		--startIt;
```

Comment:
The tentative plan with official starts seems to be to have the Dirt Belt, Deep, Paradise Worlds, and Syndicate as separate starts, with the Dirt Belt as the default. Having the list default to the end seems strange in that regard, or even with only plugin starts.

Reply:
The default-to-end was added to maintain the game's current behavior, which is that the latest-defined `start` controls how the player starts. It's arguably unexpected though, so I've removed it.